Import jeopardy.csv and create a dataframe from the data. Find out the following:

In [2]:
import pandas as pd 
import numpy as np
df = pd.read_csv("JEOPARDY_CSV.csv")
df

Show Number    Air Date             Round  \
0              4680  12/31/2004         Jeopardy!   
1              4680  12/31/2004         Jeopardy!   
2              4680  12/31/2004         Jeopardy!   
3              4680  12/31/2004         Jeopardy!   
4              4680  12/31/2004         Jeopardy!   
...             ...         ...               ...   
216925         4999   5/11/2006  Double Jeopardy!   
216926         4999   5/11/2006  Double Jeopardy!   
216927         4999   5/11/2006  Double Jeopardy!   
216928         4999   5/11/2006  Double Jeopardy!   
216929         4999   5/11/2006   Final Jeopardy!   

                               Category    Value  \
0                               HISTORY    $200    
1       ESPN's TOP 10 ALL-TIME ATHLETES    $200    
2           EVERYBODY TALKS ABOUT IT...    $200    
3                      THE COMPANY LINE    $200    
4                   EPITAPHS & TRIBUTES    $200    
...                                 ...      ...   
216925                   RIDDLE ME THIS  $2,000    
216926                        "T" BIRDS  $2,000    
216927           AUTHORS IN THEIR YOUTH  $2,000    
216928                       QUOTATIONS  $2,000    
216929                   HISTORIC NAMES     None   

                                                 Question  \
0       For the last 8 years of his life, Galileo was ...   
1       No. 2: 1912 Olympian; football star at Carlisl...   
2       The city of Yuma in this state has a record av...   
3       In 1963, live on "The Art Linkletter Show", th...   
4       Signer of the Dec. of Indep., framer of the Co...   
...                                                   ...   
216925  This Puccini opera turns on the solution to 3 ...   
216926  In North America this term is properly applied...   
216927  In Penny Lane, where this "Hellraiser" grew up...   
216928  From Ft. Sill, Okla. he made the plea, Arizona...   
216929  A silent movie title includes the last name of...   

                                Answer  
0                           Copernicus  
1                           Jim Thorpe  
2                              Arizona  
3                           McDonald's  
4                           John Adams  
...                                ...  
216925                        Turandot  
216926                      a titmouse  
216927                    Clive Barker  
216928                        Geronimo  
216929  Grigori Alexandrovich Potemkin  

[216930 rows x 7 columns]

How many rows are there in the dataframe? What does a row represent in this
dataset? What is the “grain”?


There are 216930 rows and a row represents a jeopardy question , answer , show number,category, 
value and other relevant information.

Clean the data. How do you need to “clean” it? What is “dirty” about this data?

If we look at the data types it's quiet evident that date column and value column are also showing object data types so its necessary to convert them into proper data types .
For Air Date the converted datatype would datetime64[ns]  and for value it would be float. And also,some column names have spaces , special characters so i have replaced them as well.

In [3]:
#For removing column spaces 
df.columns = df.columns.str.replace(' ', '')
print(df.columns)

Index(['ShowNumber', 'AirDate', 'Round', 'Category', 'Value', 'Question',
       'Answer'],
      dtype='object')


In [4]:
import datetime
import pandas as pd
import numpy as np
import random
df["AirDate"] = pd.to_datetime(df["AirDate"])
df["AirDate"].head()

0   2004-12-31
1   2004-12-31
2   2004-12-31
3   2004-12-31
4   2004-12-31
Name: AirDate, dtype: datetime64[ns]

In [20]:
df.dtypes

ShowNumber             int64
AirDate       datetime64[ns]
Round                 object
Category              object
Value                float64
Question              object
Answer                object
Year                   int64
Month                  int64
dtype: object

How many unique shows were there each year?


In [7]:
df['Year'] = pd.DatetimeIndex(df['AirDate']).year
df.head()
group_names = df.groupby("Year")
group_names["ShowNumber"].nunique()

Year
1984     23
1985     16
1986     25
1987     22
1988     22
1989     36
1990     73
1991     24
1992     32
1993     35
1994     19
1995     19
1996     83
1997    221
1998    223
1999    229
2000    225
2001    203
2002    114
2003    158
2004    220
2005    225
2006    230
2007    231
2008    232
2009    226
2010    231
2011    224
2012     19
Name: ShowNumber, dtype: int64

How many times was each category used?


In [8]:
df["Category"].value_counts()

BEFORE & AFTER                     547
SCIENCE                            519
LITERATURE                         496
AMERICAN HISTORY                   418
POTPOURRI                          401
                                  ... 
VIVE LA FRANCE!                      1
BUSINESS TRADEMARKS                  1
INDUSTRIALISTS                       1
MEDIEVAL DRAMA                       1
AFI's 50 GREATEST FILM VILLAINS      1
Name: Category, Length: 27983, dtype: int64

Which month (over the years) has had the most shows?

In [9]:
#create a column for the month aired
df["Month"] = pd.DatetimeIndex(df['AirDate']).month
df


ShowNumber    AirDate             Round  \
0             4680 2004-12-31         Jeopardy!   
1             4680 2004-12-31         Jeopardy!   
2             4680 2004-12-31         Jeopardy!   
3             4680 2004-12-31         Jeopardy!   
4             4680 2004-12-31         Jeopardy!   
...            ...        ...               ...   
216925        4999 2006-05-11  Double Jeopardy!   
216926        4999 2006-05-11  Double Jeopardy!   
216927        4999 2006-05-11  Double Jeopardy!   
216928        4999 2006-05-11  Double Jeopardy!   
216929        4999 2006-05-11   Final Jeopardy!   

                               Category  Value  \
0                               HISTORY    NaN   
1       ESPN's TOP 10 ALL-TIME ATHLETES    NaN   
2           EVERYBODY TALKS ABOUT IT...    NaN   
3                      THE COMPANY LINE    NaN   
4                   EPITAPHS & TRIBUTES    NaN   
...                                 ...    ...   
216925                   RIDDLE ME THIS    NaN   
216926                        "T" BIRDS    NaN   
216927           AUTHORS IN THEIR YOUTH    NaN   
216928                       QUOTATIONS    NaN   
216929                   HISTORIC NAMES    NaN   

                                                 Question  \
0       For the last 8 years of his life, Galileo was ...   
1       No. 2: 1912 Olympian; football star at Carlisl...   
2       The city of Yuma in this state has a record av...   
3       In 1963, live on "The Art Linkletter Show", th...   
4       Signer of the Dec. of Indep., framer of the Co...   
...                                                   ...   
216925  This Puccini opera turns on the solution to 3 ...   
216926  In North America this term is properly applied...   
216927  In Penny Lane, where this "Hellraiser" grew up...   
216928  From Ft. Sill, Okla. he made the plea, Arizona...   
216929  A silent movie title includes the last name of...   

                                Answer  Year  Month  
0                           Copernicus  2004     12  
1                           Jim Thorpe  2004     12  
2                              Arizona  2004     12  
3                           McDonald's  2004     12  
4                           John Adams  2004     12  
...                                ...   ...    ...  
216925                        Turandot  2006      5  
216926                      a titmouse  2006      5  
216927                    Clive Barker  2006      5  
216928                        Geronimo  2006      5  
216929  Grigori Alexandrovich Potemkin  2006      5  

[216930 rows x 9 columns]

In [17]:
monthly_data = df.groupby(["Year", "Month"])
ShowNumber = monthly_data["ShowNumber"].nunique()

Dict = {
    "ShowNumber" :ShowNumber
}

month_over_years = pd.DataFrame(Dict)

month_over_years.set_index = None
month_over_years
#group_names[["Month" , "ShowNumber"]].nunique()

ShowNumber
Year Month            
1984 9              11
     11              4
     12              8
1985 2               1
     9               1
...                ...
2011 9              10
     10             21
     11             22
     12             22
2012 1              19

[246 rows x 1 columns]

Are there any answers that appeared multiple times? Find the questions for the top
three most common answers.


In [11]:
#answers that appeared multiple times
df['Answer'].value_counts()


#some_values = ["China" , "Australia" , "Japan"]
#df.loc[df['Answer'].isin(some_values)].head(3)

China                                             216
Australia                                         215
Japan                                             196
Chicago                                           194
France                                            193
                                                 ... 
Fiber optic                                         1
estates                                             1
a mass of the dead (requiem)                        1
Bill Clinton & Al Gore                              1
Orson Welles (he was married to Rita Hayworth)      1
Name: Answer, Length: 88236, dtype: int64

In [28]:
some_values = ["China" , "Australia" , "Japan"]
df.loc[df['Answer'].isin(some_values)].head(3)

ShowNumber    AirDate             Round               Category  Value  \
404        2825 1996-12-06  Double Jeopardy!                HISTORY    NaN   
560        3036 1997-11-10         Jeopardy!  AT THE KENNEDY CENTER    NaN   
598        4107 2002-06-11         Jeopardy!      REMEMBER THE TIME    NaN   

                                              Question     Answer  Year  Month  
404  In February 1904 this country attacked the Rus...      Japan  1996     12  
560  A 1994 festival honoring this country featured...  Australia  1997     11  
598  In 1958 this country launched its second 5-yea...      China  2002      6

Visualization

In [ ]:
df["Value"].plot(kind = 'bar')